components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -618,14 +622,16 @@ protected void processAuthenticationResponse(HttpServletRequest request,
         if (StringUtils.isNotBlank(captchaParamString)) {
             context.setProperty(FrameworkConstants.CAPTCHA_PARAM_STRING, captchaParamString);
         }
-
+        Map<String, String> runtimeParams = getRuntimeParams(context);
         String loginIdentifierFromRequest = request.getParameter(USER_NAME);
+        if (StringUtils.isBlank(loginIdentifierFromRequest)) {
```

Comment:
shall we add a param to the context at the place when we are 1st checking whether the [params are sent from the runtime params](https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/200/files#diff-ce6881c48c969cee056905b6d848e6c6f828e21b62a3439bd191a6625eaa836dR201) to mention that the authentication is happening in a certain mode (i.e. params via script) and then only check for runtime params.

Reply:
Fixed